gensim/test/test_logentropy_model.py

```diff
@@ -19,9 +19,26 @@
 
 
 class TestLogEntropyModel(unittest.TestCase):
+    TEST_CORPUS = [[(1, 1.0)], [], [(0, 0.5), (2, 1.0)], []]
+
     def setUp(self):
         self.corpus_small = MmCorpus(datapath('test_corpus_small.mm'))
         self.corpus_ok = MmCorpus(datapath('test_corpus_ok.mm'))
+        self.corpus_empty = []
+
+    def test_generator_fail(self):
+        """Test creating a model using a generator as input; should fail.
+        """
+        def get_generator(test_corpus=TestLogEntropyModel.TEST_CORPUS):
+            for test_doc in test_corpus:
+                yield test_doc
+        
+        self.assertRaises(ValueError, logentropy_model.LogEntropyModel, get_generator())
```

Comment:
Can you use keyword arguments (`corpus=X`) instead of positional arguments? Looks safer, we don't want to rely on argument order (which can even change in the future…).